src/ethers/Wallet.ts

```diff
@@ -0,0 +1,49 @@
+import { ethers } from 'ethers'
+import { DEFAULT_FORNO_URL } from 'src/config'
+import { Chain } from 'src/ethers/types'
+
+export const providerUrlForChain = {
+  [Chain.Celo]: DEFAULT_FORNO_URL,
+}
+
+/**
+ * A wallet class for signing, making transactions, etc on the ethers.js platform
+ * One wallet per address per chain
+ */
+export default class Wallet extends ethers.Wallet {
+  connect(_provider: ethers.Provider | null): ethers.Wallet {
```

Comment:
Its just to be extra sure that we don't call `connect` on this wallet class. Mainly to ensure that we don't end up spawning new `ethers.Wallet` s that are similar but not exactly like `Wallet` and may not properly get destroyed by the locking mechanism in KeychainAccountManager

Reply:
Another way to ensure this would be to just have our `Wallet` _contain_ an `ethers.Wallet` rather than extending one (e.g. composition vs. inheritance) though not sure if there are wider consequences there re: how `Wallet` is used externally.

Reply:
I actually originally had an implementation where I used composition like you said. At this point in the development I can't see a strong case in either direction, but I'd definitely be open to it if during early development we see some other advantages for composition.

Reply:
Joe's suggestion to use composition sounds preferable because we'd get a compile-time error rather than a runtime error for this undesirable behavior